include/fiction/utils/math_utils.hpp

```diff
@@ -76,6 +79,45 @@ T integral_abs(const T n) noexcept
     return result;
 }
 
+/**
+ * This function generates all possible combinations of distributing k entities
+ * onto n positions. Each combination is represented as a vector of indices,
+ * where each index indicates the position of an entity.
+ *
+ * @param k The number of entities to distribute.
+ * @param n The number of positions available for distribution.
+ * @return A vector of vectors representing all possible combinations of
+ *         distributing k entities on n positions.
+ */
+[[nodiscard]] inline std::vector<std::vector<std::size_t>>
+determine_all_combinations_of_distributing_k_entities_on_n_positions(const uint64_t k, const uint64_t n) noexcept
```

Comment:
Is there a reason you need to allocate the memory for the entire vector of vectors instead of working with the iterators of `combinations.h` directly?

Reply:
I think, mutithreading for gate designing is a little bit easier, right?

Reply:
True, that's a good use case.